resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocatorBlacklistTracker.scala

```diff
@@ -103,7 +117,32 @@ private[spark] class YarnAllocatorBlacklistTracker(
     refreshBlacklistedNodes()
   }
 
-  def isAllNodeBlacklisted: Boolean = currentBlacklistedYarnNodes.size >= numClusterNodes
+  def isAllNodeBlacklisted: Boolean = {
+    refreshYarnNodes()
+    val allBlacklisted = allRunningNodes.diff(currentBlacklistedYarnNodes).isEmpty
```

Comment:
The `allRunningNodes` is not needed as a separate member `var`: you can do the filtering from `allNodes` here. 

This way you save its calculation at the `excludeNodes()` def where the `allRunningNodes` is not needed.